.github/workflows/_supergraph_publish.yaml

```diff
@@ -15,45 +15,80 @@ jobs:
         with:
           fetch-depth: 0
 
+      - name: Setup Node
+        uses: actions/setup-node@v4.1.0
+
+      - name: Install release-please
+        run: npm install release-please simple-git
+
+      - name: Generate Chart Version
+        id: version
+        uses: actions/github-script@v7.0.1
+        with:
+          result-encoding: string
+          script: |
+            const releasePlease = require('release-please');
+            const simpleGit = require('simple-git');
+
+            const ghRelease = await releasePlease.GitHub.create({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              token: '${{ github.token }}'
+            });
+
+            const manifest = await releasePlease.Manifest.fromManifest(
+              ghRelease,
+              ghRelease.repository.defaultBranch,
+              'release-please-config.json',
+              '.release-please-manifest.json',
+              {}
+            );
+
+            const pullRequests = await manifest.buildPullRequests();
+            const pullRequest = pullRequests.find((pullRequest) => pullRequest.updates.some((update) => update.path === 'charts/supergraph/Chart.yaml'));
+            console.log(`Supergraph Pull Request: ${JSON.stringify(pullRequest)}`);
+
+            const git = simpleGit();
+            const lastTag = await git.raw('describe', '--tags', '--match', 'supergraph@v*', 'HEAD').catch((err) => undefined);
+            console.log(`Last Tag: ${JSON.stringify(lastTag)}`);
+            const commitsSince = lastTag ? await git.raw('rev-list', `${lastTag}..HEAD`, '--count') : await git.raw('rev-list', '--count', '--all');
+            console.log(`Commits Since: ${commitsSince}`);
+
+            const releaseData = pullRequest.body.releaseData.find((release) => release.component === 'supergraph-schema');
+            const rcVersion = pullRequest ? `${releaseData.version.major}.${releaseData.version.minor}.${releaseData.version.patch}-rc${commitsSince}` : '';
+            console.log(`Release Candidate Version: ${rcVersion}`);
+            return rcVersion
```

Comment:
This is quite a lot of code. Could it be extracted into a separate script? https://github.com/actions/github-script?tab=readme-ov-file#run-a-separate-file

Reply:
Good call, done

Reply:
In the process found a bug where the tag would be wrong if there were no new updates to the schema